### DISCUSSION TITLE
Build Question

### DISCUSSION THREAD

**Discussion by elosoman** [Ultrawipf/OpenFFBoard#40] [General]
I just uploaded all the board info to pcbway and clicked the order button. Might you know anything about the following request?

Dear Customer, Thank you for your PCBA order [] , we checked all the files you uploaded and find there is no pick&place(centroid) file. Please go to 'Production Status' to upload this file . If you really can't provide this file, please contact your service representative to explain that.

I confess this is my first time going through the board production process. Might you have a clue on what this is or do i tell them i don't have one.

Thanks

**Comment by Ultrawipf** [comment]
You have to generate the placement file with eagle and edit it manually for their order process. Every service requires different formats.
The board does require some thoughts and possibly different component choices depending on the application and supply situation so thats mostly up to you how to build it.

I would even recommend not making it in the current situation and maybe waiting for a final version as there will be further changes and many parts in that version are currently very hard to get.

**Reply by elosoman** [reply]
well we shall see ... pcbway got back to me and they said they where able to work around the file. It should be interesting ... here are the pics that they sent me. The order is apparently shipping so cross your fingers.
![T-1N2W337306A-02](https://user-images.githubusercontent.com/1059072/139699379-9c81e92e-bb7f-472e-ab1d-b90fae802a3c.jpg)
![T-1N2W337306A-01](https://user-images.githubusercontent.com/1059072/139699384-f1800097-8075-41aa-bc2c-be32cf575c31.jpg)

Thanks!

**Reply by elosoman** [reply]
So I received my lot of boards as pictured above from pcbway. Any chance you could upload a reference picture for verification?

**Reply by elosoman** [reply]
So it looks as if the board looks good. I'm getting a power led when connected via usb. The STM IDE is able to build all of the current master branch. But is failing as pictured. No ST-LINK detected.
![Screen Shot 2021-11-03 at 11 42 30 PM](https://user-images.githubusercontent.com/1059072/140254417-6be2033e-f0ea-4454-857b-cd3453fea946.png)
 
Is a ST-LINK required to boot the firmware. I also found STM32CubeProgrammer but it is unable to find a device on USB. Any clues would be appreciated. Apologies if this is not the right place for this topic.

**Reply by jonlidgard** [reply]
Best to check on Discord as it was a while a go but I think you can download the firmware over USB. Use CubeProgrammer with the image you generated in the IDE. You have to short out Boot0 on the board on power up to put it in DFU mode.

**Reply by elosoman** [reply]
thanks for the tip @jonlidgard ... got the firmware loaded. Configurator can see the board and connects but its just the terminal and system parts seem to work. i'm not seeing any of the additional interface ... is that because no motor is connected?

![Screen Shot 2021-11-05 at 7 26 50 PM](https://user-images.githubusercontent.com/1059072/140588868-0081ba2b-c13f-4c54-bf2a-b180b8782238.png)
.

**Reply by Ultrawipf** [reply]
You need to select the FFB Wheel Main Class. "Basic" is a failsafe mode. 
The firmware is designed that it can dynamically load different main programs without reflashing.

**Reply by elosoman** [reply]
That was my thought from looking at mainclass_chooser.cpp ... I'm not seeing any change when selecting a new Main Class ... clicking the change button causes the board to reboot. After boot it comes back to Basic.

![Screen Shot 2021-11-06 at 11 13 36 AM](https://user-images.githubusercontent.com/1059072/140614556-bbbd22af-69e6-46c5-9e64-5fa7250b860a.png)

**Reply by Ultrawipf** [reply]
Ah hm. Try reflashing and do a full chip erase. Its a rare bug with the eeprom emulation not writing into the flash. Hard to reproduce but a full chip erase normally fixes it.

**Reply by elosoman** [reply]
good to know ... that did the trick!
